HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+Eigen::Matrix3d rpy2rotmat(const Eigen::Array3d &rollPitchYaw, const RotationConvention &rotation);
+std::string toString(RotationConvention convention);
+void compareRotation(const Eigen::Matrix3d &rotationMatrix1, const Eigen::Matrix3d &rotationMatrix2);
+void compareRotationExtended(const Eigen::Array3d &rollPitchYaw,
+                             const Eigen::Matrix3d &rotationMatrix,
+                             const RotationConvention &rotation);
+Representations transformFromZividToRobot(const Eigen::Matrix3d &);
+void transformFromRobotToZivid(const Eigen::Matrix3d &, const Representations &);
+
+int main()
+{
+    try
+    {
+        std::cout << "This example shows transformations to/from Transformation Matrix" << std::endl;
+
+        std::cout << "Opening .YAML file which contains transformation matrix (PoseState node)" << std::endl;
+        auto fileStorageIn = cv::FileStorage{};
+        if(!fileStorageIn.open("..\\robotTransform.yaml", cv::FileStorage::Mode::READ))
+        {
+            throw std::runtime_error("Could not open ..\\robotTransform.yaml");
+        }
+        const auto poseStateNode = fileStorageIn["PoseState"];
+        std::cout << "Getting PoseState node" << std::endl;
+        if(poseStateNode.empty())
+        {
+            fileStorageIn.release();
+            throw std::runtime_error("PoseState node not found in file");
+        }
+        const auto poseState = poseStateNode.mat();
+        std::cout << poseState << std::endl;
+        fileStorageIn.release();
+
+        // Extract Rotation Matrix and Translation Vector from Transformation Matrix
+        std::cout << "\nConverting transformation matrix to rotation matrix and translation vector" << std::endl;
+        const Eigen::Matrix3d rotationMatrix = cvToEigen(poseState(cv::Range(0, 3), cv::Range(0, 3)));
+        const Eigen::Vector3d translationVector = cvToEigen(poseState(cv::Range(0, 3), cv::Range(3, 4)));
+        std::cout << "\nRotationMatrix:\n" << rotationMatrix << std::endl;
+        std::cout << "\nTranslationVector:\n" << translationVector << std::endl;
+
+        // Transform from Zivid to Robot (Rotation Matrix --> any format)
+        const auto robotRepresentations = transformFromZividToRobot(rotationMatrix);
+
+        // Transform from Robot to Zivid (any format --> Rotation Matrix)
+        transformFromRobotToZivid(rotationMatrix, robotRepresentations);
+    }
+
+    catch(const std::exception &e)
+    {
+        std::cerr << "Error: " << e.what() << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+cv::Mat eigenToCv(const Eigen::MatrixXd &eigenMat)
+{
+    cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+    for(int i = 0; i < cvMat.rows; i++)
+    {
+        for(int j = 0; j < cvMat.cols; j++)
+        {
+            cvMat.at<double>(i, j) = eigenMat(i, j);
+        }
+    }
+
+    return cvMat;
+}
+
+Eigen::MatrixXd cvToEigen(const cv::Mat &cvMat)
```

Comment:
You can use `opencv::eigen2cv()` (https://docs.opencv.org/3.4.8/d3/ddd/eigen_8hpp.html)